model/mobile_trigger.go

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package model
+
+import (
+	"encoding/json"
+	"io"
+)
+
+type MobileTrigger struct {
+	PluginId string `json:"id"`
+	Location string `json:"location"`
```

Comment:
My opinion might be caused by understand for the English language: To me `Location` is a fixed place. But the trigger might be shown in different locations based on the used client. `Type` captures more that it's about "what" kind of Trigger.

Reply:
Probably the problem is calling Trigger to this object. I changed it to MobileIntegration. Each integration has a location.

Reply:
I changed it again to PluginIntegration, to allow scopes.